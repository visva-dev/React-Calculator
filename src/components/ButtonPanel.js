import React from 'react';
import Button from './Button';

const ButtonPanel = () => {
  const firstGroup = ['AC', '+/-', '%', '/'];
  const secondGroup = ['7', '8', '9', 'X'];
  const thirdGroup = ['4', '5', '6', '-'];
  const fourthGroup = ['1', '2', '3', '+'];
  const fifthGroup = ['0', '.', '='];

  return (
    <>
      <div className="group1">
        {firstGroup.map(element => (
          <Button name={element} key={firstGroup.indexOf(element)} />
        ))}
      </div>
      <div className="group1">
        {secondGroup.map(element => (
          <Button name={element} key={secondGroup.indexOf(element)} />
        ))}
      </div>
      <div className="group1">
        {thirdGroup.map(element => (
          <Button name={element} key={thirdGroup.indexOf(element)} />
        ))}
      </div>
      <div className="group1">
        {fourthGroup.map(element => (
          <Button name={element} key={fourthGroup.indexOf(element)} />
        ))}
      </div>
      <div className="group1">
        {fifthGroup.map(element => (
          <Button name={element} key={fifthGroup.indexOf(element)} />
        ))}
      </div>
    </>
  );
};

export default ButtonPanel;
