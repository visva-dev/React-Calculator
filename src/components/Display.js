import PropTypes from 'prop-types';

const Display = props => {
  const { result } = props;
  return (
    <>
      {result}
    </>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: 'O',
};

export default Display;
