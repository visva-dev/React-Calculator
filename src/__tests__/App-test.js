import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/Navbar';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <div className="app">
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
        <Display />
        <ButtonPanel />
      </div>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
