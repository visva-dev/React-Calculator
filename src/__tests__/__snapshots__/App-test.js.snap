// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/App"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="Display"
  >
    O
  </div>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="%"
    >
      %
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "10%",
        }
      }
      type="button"
      value="/"
    >
      /
    </button>
  </div>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="9"
    >
      9
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "10%",
        }
      }
      type="button"
      value="X"
    >
      X
    </button>
  </div>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="6"
    >
      6
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "10%",
        }
      }
      type="button"
      value="-"
    >
      -
    </button>
  </div>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="3"
    >
      3
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "10%",
        }
      }
      type="button"
      value="+"
    >
      +
    </button>
  </div>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "20%",
        }
      }
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "width": "10%",
        }
      }
      type="button"
      value="."
    >
      .
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "10%",
        }
      }
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
